Sources/SourceKitLSP/Swift/CodeActions/ConvertStringConcatenationToStringInterpolation.swift

```diff
@@ -0,0 +1,184 @@
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+import SwiftRefactor
+import SwiftSyntax
+
+/// ConvertStringConcatenationToStringInterpolation is a code action that converts a valid string concatenation into a
+/// string interpolation.
+struct ConvertStringConcatenationToStringInterpolation: SyntaxRefactoringProvider {
+  static func refactor(syntax: SequenceExprSyntax, in context: Void) -> SequenceExprSyntax? {
+    guard let (componentsOnly, commonPounds) = preflight(exprList: syntax.elements) else {
+      return nil
+    }
+
+    var ret: StringLiteralSegmentListSyntax = []
+    for component in componentsOnly {
+      guard let stringLiteral = component.as(StringLiteralExprSyntax.self) else {
+        ret.append(
+          .expressionSegment(
+            ExpressionSegmentSyntax(
+              pounds: commonPounds,
+              expressions: [
+                LabeledExprSyntax(expression: component.trimmed)
+              ]
+            )
+          )
+        )
+        continue
+      }
+
+      var segments = stringLiteral.segments
+      if let commonPounds, stringLiteral.isPoundsNotEqual(to: commonPounds) {
+        for i in segments.indices {
+          guard case let .expressionSegment(exprSegment) = segments[i] else {
+            continue
+          }
+          segments[i] = .expressionSegment(exprSegment.with(\.pounds, commonPounds))
+        }
+      }
+      ret += segments
+    }
+
+    return syntax.with(
+      \.elements,
+      [
+        ExprSyntax(
+          StringLiteralExprSyntax(
+            openingPounds: commonPounds,
+            openingQuote: .stringQuoteToken(),
+            segments: ret,
+            closingQuote: .stringQuoteToken(),
+            closingPounds: commonPounds
+          )
+        )
+      ]
+    )
+  }
+
+  /// If `exprList` is a valid string concatenation, returns 1) all elements in `exprList` with concat operators
+  /// stripped and 2) the longest pounds amongst all string literals, otherwise returns nil.
+  ///
+  /// `exprList` as a valid string concatenation must contain n >= 3 children where n is an odd number with a concat
+  /// operator `+` separating every other child, which must either be a single-line string literal or a valid
+  /// expression for string interpolation. `exprList` must also contain at least one string literal child.
+  ///
+  ///  The following is a valid string concatenation.
+  ///``` swift
+  ///"Hello " + aString + "\(1)World"
+  ///```
+  ///  The following are invalid string concatenations.
+  ///``` swift
+  ///aString + bString // no string literals
+  ///
+  ///"Hello " * aString - "World" // non `+` operators
+  ///
+  ///"""
+  ///Hello
+  ///"""
+  ///+ """
+  ///World
+  ///""" // multi-line string literals
+  ///```
```

Comment:
Seems like you’re missing some whitespace for the code blocks to make them look nice.

```suggestion
  /// "Hello " + aString + "\(1)World"
  /// ```
  ///  The following are invalid string concatenations.
  /// ``` swift
  /// aString + bString // no string literals
  ///
  /// "Hello " * aString - "World" // non `+` operators
  ///
  /// """
  /// Hello
  /// """
  /// + """
  /// World
  /// """ // multi-line string literals
  /// ```
```